src/libraries/System.Security.Cryptography.Cose/tests/CoseTestKeyManager.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Test.Cryptography;
+using Xunit;
+
+#pragma warning disable SYSLIB5006
+
+namespace System.Security.Cryptography.Cose.Tests
+{
+    public sealed class CoseTestKeyManager : IDisposable
+    {
+        public const string RSAPkcs1Identifier = "RSA-PKCS1";
+        public const string RSAPssIdentifier = "RSA-PSS";
+        public const string ECDsaIdentifier = "ECDsa";
+        public const string MLDSA44Identifier = "ML-DSA-44";
+        public const string MLDSA65Identifier = "ML-DSA-65";
+        public const string MLDSA87Identifier = "ML-DSA-87";
+
+        private Dictionary<string, CoseTestKey> _coseKeys = new();
+
+        // Ideally public keys should be used with verify, i.e.:
+        // - CoseTestKeyManager.ctor() => private keys
+        // - CoseTestKeyManager CreateTestPublicKeys(CoseTestKeyManager privateKeys)
+        // this is ok for testing purposes.
+        private CoseTestKeyManager() { }
+
+        public static CoseTestKeyManager CreateTestKeys()
+        {
+            CoseTestKeyManager keyManager = new();
+
+            WithTestKeysInfo((keyId, keyType, hashAlgorithm) =>
+            {
+                keyManager.AddKey(keyId, keyType, hashAlgorithm);
+            });
+
+            return keyManager;
+        }
+
+        public static string[] GetAllKeyIds()
+        {
+            List<string> keyIds = new();
+            WithTestKeysInfo((keyId, _, _) =>
+            {
+                keyIds.Add(keyId);
+            });
+
+            return keyIds.ToArray();
+        }
+
+        private static void WithTestKeysInfo(Action<string, IDisposable, CoseKey> action)
+        {
+            Let(ECDsa.Create(), (key) => action(ECDsaIdentifier, key, CoseKey.FromKey(key, HashAlgorithmName.SHA256)));
+            Let(RSA.Create(), (key) => action(RSAPkcs1Identifier, key, CoseKey.FromKey(key, RSASignaturePadding.Pkcs1, HashAlgorithmName.SHA256)));
+
+            if (PlatformSupport.IsRsaPssSupported)
+            {
+                Let(RSA.Create(), (key) => action(RSAPssIdentifier, key, CoseKey.FromKey(key, RSASignaturePadding.Pss, HashAlgorithmName.SHA256)));
+            }
+
+            if (MLDsa.IsSupported)
+            {
+                Let(MLDsa.GenerateKey(MLDsaAlgorithm.MLDsa44), (key) => action(MLDSA44Identifier, key, CoseKey.FromKey(key)));
+                Let(MLDsa.GenerateKey(MLDsaAlgorithm.MLDsa65), (key) => action(MLDSA65Identifier, key, CoseKey.FromKey(key)));
+                Let(MLDsa.GenerateKey(MLDsaAlgorithm.MLDsa87), (key) => action(MLDSA87Identifier, key, CoseKey.FromKey(key)));
+            }
+
+            static void Let<KeyType>(KeyType key, Action<KeyType> action) => action(key);
```

Comment:
Generic parameters should start with a T, makes it easier to read when something is still generic.

```suggestion
            static void Let<TKeyType>(TKeyType key, Action<TKeyType> action) => action(key);
```